yetanotherco/aligned_layer
# feat: add proving system to vk or program code commitment

This PR changes how the verification key commitment is calculated. Now the commitment is made as follows,
```
commitment = keccak256(verification_key_bytes | proving_system_id)
```

This is needed to avoid attacks where verification keys from different proving systems could be used in a verification.

## Testing
* Send proofs as usual and check everything works as expected
* Test the `get-vk-commitment` as follows:
    * Add this to line 201 of `batcher/aligned-sdk/src/sdk.rs`:
  
    ```rust
    println!("Verification key commitment: {:?}", hex::encode(aligned_verification_data[0].verification_data_commitment.proving_system_aux_data_commitment));
    ```
    * Run `make install_aligned_compiling`
    * At the root of the repo, run
    ```
    aligned get-vk-commitment --verification_key_file scripts/test_files/gnark_plonk_bn254_script/plonk.vk --proving_system GnarkPlonkBn254
    ```
    The commitment will be logged in the terminal.
    * Send some plonk proofs using `make batcher_send_plonk_bn254_burst` and see that the printed commitment is the same as the one you got with the `get-vk-commitment` command.
